snappy/snapp.go

```diff
@@ -1936,3 +1953,40 @@ func makeSnapHookEnv(part *SnapPart) (env []string) {
 
 	return env
 }
+
+func (m *packageYaml) addSquashfsMount(baseDir string, inhibitHooks bool, inter interacter) error {
+	squashfsPath := stripGlobalRootDir(squashfs.BlobPath(baseDir))
+	whereDir := stripGlobalRootDir(baseDir)
+
+	sysd := systemd.New(dirs.GlobalRootDir, inter)
+	mountUnitName, err := sysd.WriteMountUnitFile(m.Name, squashfsPath, whereDir)
+	if err != nil {
+		return err
+	}
+
+	// we always enable the mount unit even in inhibit hooks
+	if err := sysd.Enable(mountUnitName); err != nil {
+		return err
+	}
+
+	if !inhibitHooks {
+		return sysd.Start(mountUnitName)
+	}
+
+	return nil
+}
+
+func (m *packageYaml) removeSquashfsMount(baseDir string, inter interacter) error {
+	sysd := systemd.New(dirs.GlobalRootDir, inter)
+	unit := systemd.MountUnitPath(stripGlobalRootDir(baseDir), "mount")
+	if helpers.FileExists(unit) {
+		// we ignore errors, nothing should stop removals
```

Comment:
Should we return errors we get, though? Not stopping feels sane, but feels like we should still let the user know something is not quite right.